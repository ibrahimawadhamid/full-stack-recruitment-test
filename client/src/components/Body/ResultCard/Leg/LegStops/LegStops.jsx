import React, {Component} from 'react';
import BpkPopover from 'bpk-component-popover';
import {BpkTable, BpkTableBody, BpkTableCell, BpkTableHead, BpkTableHeadCell, BpkTableRow,} from 'bpk-component-table';
import moment from 'moment';

class LegStops extends Component {
  render() {
    return (
        <div id={this.props.itineraryId + this.props.legInfo["Id"]}>
          <BpkPopover id={this.props.itineraryId + this.props.legInfo["Id"] + "more-stops-popover"}
            target={() => document.getElementById(this.props.itineraryId + this.props.legInfo["Id"]+"moreStops")}
            onClose={this.props.closeAdditionalStopsPopover}
            isOpen={this.props.isAdditionalStopsOpen}
            label="More Stops popover"
            closeButtonText="Close"
            renderTarget={() =>
              document.getElementById(this.props.itineraryId + this.props.legInfo["Id"])
            }
            closeButtonProps={{
              tabIndex: 0,
            }}>
            <BpkTable>
              <BpkTableHead>
                <BpkTableRow>
                  <BpkTableHeadCell>From</BpkTableHeadCell>
                  <BpkTableHeadCell>To</BpkTableHeadCell>
                  <BpkTableHeadCell>Duration</BpkTableHeadCell>
                </BpkTableRow>
              </BpkTableHead>
              <BpkTableBody>
            {this.props.legInfo["Segments"].map(segment => {
              return (
                <BpkTableRow key={segment["Id"]}>
                  <BpkTableCell>{segment["OriginCode"] + " " + moment(segment["DepartureDateTime"]).format("HH:MM")}</BpkTableCell>
                  <BpkTableCell>{segment["DestinationCode"] + " " + moment(segment["ArrivalDateTime"]).format("HH:MM")}</BpkTableCell>
                  <BpkTableCell>{moment.utc(segment["Duration"]*60000).format("H[h] mm")}</BpkTableCell>
                </BpkTableRow>
              )
            })}
              </BpkTableBody>
            </BpkTable>
          </BpkPopover>
        </div>
    );
  };
}

export default LegStops;
