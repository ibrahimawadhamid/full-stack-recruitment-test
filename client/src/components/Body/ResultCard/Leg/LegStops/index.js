import LegStops from './LegStops';

export default LegStops;
