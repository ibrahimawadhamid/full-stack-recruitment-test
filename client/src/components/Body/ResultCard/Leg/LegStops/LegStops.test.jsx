import React from 'react';
import TestRenderer from 'react-test-renderer';

import LegStops from './LegStops';

describe('LegStops', () => {
    


    it('should render correctly', () => {
        const tree = TestRenderer.create(
            <LegStops legInfo={{ Id: 'id', Segments: [] }}
                isAdditionalStopsOpen={false}
                closeAdditionalStopsPopover={() => { }} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});