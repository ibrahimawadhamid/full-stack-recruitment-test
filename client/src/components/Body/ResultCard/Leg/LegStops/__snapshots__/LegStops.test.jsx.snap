// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegStops should render correctly 1`] = `
<div
  id="undefinedid"
/>
`;
